tests/python/pants_test/testutils/process_test_util.py

```diff
@@ -5,21 +5,39 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+from contextlib import contextmanager
+
 import psutil
 
 
 class ProcessStillRunning(AssertionError):
   """Raised when a process shouldn't be running but is."""
 
 
-def assert_no_process_exists_by_command(name):
-  """Asserts that no process exists for a given command with a helpful error."""
+def _safe_iter_matching_processes(name):
   for proc in psutil.process_iter():
     try:
-      cmdline = proc.cmdline()
-      if name in ''.join(cmdline):
-        raise ProcessStillRunning(
-          'a {} process was detected at PID {} (cmdline={})'.format(name, proc.pid, cmdline)
-        )
+      if name in ''.join(proc.cmdline()):
+        yield proc
     except (psutil.NoSuchProcess, psutil.AccessDenied):
       pass
+
+
+def _matching_process_set(name):
+  return {p for p in _safe_iter_matching_processes(name)}
```

Comment:
`set(_safe_iter_matching_processes(name))`? Might make the helper method unnecessary...